### PR TITLE
[Storage] fix UseSubDomainName property name, and add Canfailover pro…

### PR DESCRIPTION
**Pull request by blueww**
## Description

Update SDK with fix the "UseSubDomainName" property name (it's wrong as "UseSubDomain" before), and add the latest change from swagger, include some typo fix and add "CanFailOver" property.

Since don't plan to release the change now, so not change the version.

The SDK I generated from my personal swagger repo, since the PR is still in review: https://github.com/Azure/azure-rest-api-specs/pull/4938
Will update it to official swagger after the PR merged in.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Storage/Management.Storage/Generated/Models/CustomDomain.cs
Comment (dsgouda) [inline src/SDKs/Storage/Management.Storage/Generated/Models/CustomDomain.cs]: Please note that this is potentially a breaking change

**Review by dsgouda — CHANGES_REQUESTED** [review]
Do you intend to bump the version numbers in this PR?

**Comment by blueww** [comment]
@dsgouda 

I have upgrade the version to 9.2.0-preview. 
Anyway, we might will not release 9.2.0-preview now, but release 10.0.0 stable version in the next release.

File: src/SDKs/Storage/Management.Storage/Generated/Models/CustomDomain.cs
Comment (blueww) [reply src/SDKs/Storage/Management.Storage/Generated/Models/CustomDomain.cs]: Yes, this breaking is intend. Since the before "UseSubDomain" is wrong. It should be "UseSubDomainName".

**Comment by dsgouda** [comment]
Thanks @blueww  will approve and merge once the REST spec is merged.

**Comment by blueww** [comment]
@dsgouda 

The swagger PR is merged, I have regenerated the SDK code on the merged swagger, not code change but only storage_resource-manager.txt has change.
Please help to review and merge this PR.

**Review by dsgouda — APPROVED** [review]

**Pull request merged**
